module/chat.js

```diff
@@ -17,3 +19,53 @@ export const highlightCriticalSuccessFailure = function (message, html, data) {
   if (d.total >= 20) html.find('.dice-total').addClass('critical')
   else if (d.total <= 1) html.find('.dice-total').addClass('fumble')
 }
+
+/* -------------------------------------------- */
+
+/**
+ * This function is used to hook into the Chat Log context menu to add additional options to each message
+ * These options make it easy to conveniently apply damage to controlled tokens based on the value of a Roll
+ *
+ * @param {HTMLElement} html    The Chat Message being rendered
+ * @param {Array} options       The Array of Context Menu options
+ *
+ * @return {Array}              The extended options Array including new context choices
+ */
+export const addChatMessageContextOptions = function (html, options) {
+  const canApply = li => canvas.tokens.controlled.length && li.find('.damage-applyable').length
+  options.push(
+    {
+      name: game.i18n.localize('DCC.ChatContextDamage'),
+      icon: '<i class="fas fa-user-minus"></i>',
+      condition: canApply,
+      callback: li => applyChatCardDamage(li, 1)
+    }
+  )
+  options.push(
+    {
+      name: game.i18n.localize('DCC.ChatContextHealing'),
+      icon: '<i class="fas fa-user-plus"></i>',
+      condition: canApply,
+      callback: li => applyChatCardDamage(li, -1)
+    }
+  )
+  return options
+}
+
+/* -------------------------------------------- */
+
+/**
+ * Apply rolled dice damage to the token or tokens which are currently controlled.
+ * This allows for damage to be scaled by a multiplier to account for healing, critical hits, or resistance
+ *
+ * @param {HTMLElement} roll    The chat entry which contains the roll data
+ * @param {Number} multiplier   A damage multiplier to apply to the rolled damage.
+ * @return {Promise}
+ */
+function applyChatCardDamage (roll, multiplier) {
+  const amount = roll.find('.damage-applyable').text()
```

Comment:
I found a bug with this, where if I expand the roll in the chat before applying the damage, the value in the <li> is `# = # + #` instead of just `#`, and then all the math goes crazy.  I think you may need to dig a little and use something other than .text() here.  Hopefully you won't have to regex the total out yourself. :D

Reply:
I now pass the damage through a custom attribute (data-damage) which avoids this problem and allows other chat messages to be given class damage-applyable without worrying about how the contents are formatted - just the attribute matters.

Reply:
Looks good now!